fe/fe-core/src/main/java/org/apache/doris/datasource/credentials/CredentialUtils.java

```diff
@@ -17,22 +17,69 @@
 
 package org.apache.doris.datasource.credentials;
 
-import com.google.common.annotations.VisibleForTesting;
+import org.apache.doris.datasource.property.storage.StorageProperties;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
+/**
+ * Utility class for Credential operations
+ */
 public class CredentialUtils {
+
+    /**
+     * Supported cloud storage prefixes for filtering vended credentials
+     */
+    private static final Set<String> CLOUD_STORAGE_PREFIXES = new HashSet<>(Arrays.asList(
+            "s3.",      // Amazon S3
+            "oss.",     // Alibaba OSS
+            "cos.",     // Tencent COS
+            "obs.",     // Huawei OBS
+            "gs.",      // Google Cloud Storage
+            "azure."    // Microsoft Azure
+    ));
+
+    /**
+     * Filter cloud storage properties from raw vended credentials
+     * Only keeps properties with supported cloud storage prefixes
+     *
+     * @param rawVendedCredentials Raw vended credentials map
+     * @return Filtered cloud storage properties
+     */
+    public static Map<String, String> filterCloudStorageProperties(Map<String, String> rawVendedCredentials) {
+        if (rawVendedCredentials == null || rawVendedCredentials.isEmpty()) {
+            return new HashMap<>();
+        }
+
+        Map<String, String> filtered = new HashMap<>();
+        rawVendedCredentials.entrySet().stream()
+                .filter(entry -> entry.getKey() != null && entry.getValue() != null)
+                .filter(entry -> CLOUD_STORAGE_PREFIXES.stream().anyMatch(prefix -> entry.getKey().startsWith(prefix)))
+                .forEach(entry -> filtered.put(entry.getKey(), entry.getValue()));
+
+        return filtered;
+    }
+
     /**
-     * Future method for FileIO-based credential extraction.
-     * This method signature is designed to be compatible with future FileIO implementations.
+     * Extract backend properties from StorageProperties map
+     * Reference: CatalogProperty.getBackendStorageProperties()
      *
-     * @param fileIoProperties properties from FileIO (reserved for future use)
-     * @param extractor custom credential extractor
-     * @return extracted credentials
+     * @param storagePropertiesMap Map of storage properties
+     * @return Backend properties with null values filtered out
      */
-    @VisibleForTesting
-    public static Map<String, String> extractCredentialsFromFileIO(Map<String, String> fileIoProperties,
-            CredentialExtractor extractor) {
-        return extractor.extractCredentials(fileIoProperties);
+    public static Map<String, String> getBackendPropertiesFromStorageMap(
```

Comment:
This method will be call multiple times(eg, for each split). it is very costy.
Since the input map(`storagePropertiesMap`) is initialized when creating scan node, i think this method can also be called only once when initializing the scan node